Vizzuality/half-earth-v3
# feat: new navigation menu implementation [H-531] [H-532] [HE-494]

## New navigation menu Implementation

### Designs

[Figma Prototype](https://www.figma.com/proto/2YAp3AS5lYCj2IWEAN8AJw/HE-Map---Work-in-Progress?node-id=272%3A63260&viewport=-256%2C1968%2C0.13&scaling=scale-down&starting-point-node-id=4%3A19661)

### Testing instructions
Please, add `REACT_APP_FEATURE_NEW_MENUS` feature flag to `.env`

Test if everything works fine with the feature disable. 

Once you set the feature, test:
- Feature is disable on mobile.
- Landing page should no longer have the side menu but a button that displays the "About the map" modal.
- All pages except the landing should have a right side menu with: 
       - location searcher
       - zoom controls
       - share button
       - help button which display a modal "How to navigate the map"
       - language switcher
- All main pages except the landing should have a globe menu indicator with the corresponding globe highlighted. (SEE NOTES ON USER STORY SAYING IT SHOULD BE BIGGER THAN IN THE DESIGNS)
- Both side and bottom menu should be disable on on boarding mode

### Feature user story ticket
[HE-494](https://vizzuality.atlassian.net/browse/HE-494)

### Feature relevant tickets
[HE-531](https://vizzuality.atlassian.net/browse/HE-531)
[HE-532](https://vizzuality.atlassian.net/browse/HE-532)

### Recommendations
We have two types of modals with two different overlays. The modal component is from the he-components library and you cannot conditionally override the overlay, so the component will be changed to a custom one in a future task.
